pennylane/qnode.py

```diff
@@ -613,18 +613,20 @@ def __call__(self, *args, **kwargs):
         supports_hamiltonian = self.device.supports_observable("Hamiltonian")
         hamiltonian_in_obs = "Hamiltonian" in [obs.name for obs in self.qtape.observables]
         if hamiltonian_in_obs and not supports_hamiltonian:
-            try:
-                tapes, fn = qml.transforms.hamiltonian_expand(self.qtape, group=False)
-            except ValueError as e:
+            if (
+                len(self.qtape.measurements) > 1
+                or self.qtape.measurements[0].return_type != qml.measure.Expectation
+            ):
                 raise ValueError(
-                    "At the moment, only single expectations of Hamiltonian observables can be returned"
-                    "on the {} device.".format(self.device.name)
-                ) from e

```

Comment:
Before there was no check whether the measurement type is an expectation. At the moment we don't really know how `sample` and `var` can be supported, because they are not linear in the operations.